### PR TITLE
Fix MSC2716 `/batch_send` endpoint not being able to create outlier events

### PR DESCRIPTION
**Pull request by MadLittleMods**
Fix MSC2716 `/batch_send` endpoint not being able to create outlier events.

The `EventContext.for_outlier` refactor happened in https://github.com/matrix-org/synapse/pull/10883 and this spot was left out.

Part of https://github.com/matrix-org/matrix-doc/pull/2716

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [ ] ~~Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)~~
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: synapse/handlers/message.py [hunk 1]
```diff
@@ -953,18 +953,13 @@ async def create_new_client_event(
             depth=depth,
         )
 
-        old_state = None
+        context = await self.state.compute_event_context(event)
```
Comment (richvdh) [inline synapse/handlers/message.py:956]: computing event context isn't free (eg, it creates state groups in the database). We shouldn't do it if we're about to throw it away.

File: synapse/handlers/message.py [hunk 2]
```diff
@@ -953,18 +953,13 @@ async def create_new_client_event(
...
 
         # Pass on the outlier property from the builder to the event
         # after it is created
         if builder.internal_metadata.outlier:
             event.internal_metadata.outlier = builder.internal_metadata.outlier
```
Comment (richvdh) [inline synapse/handlers/message.py:961]:
```suggestion
            event.internal_metadata.outlier = True
```

File: changelog.d/10938.bugfix [hunk 3]
```diff
@@ -0,0 +1 @@
+Fix [MSC2716](https://github.com/matrix-org/matrix-doc/pull/2716) `/batch_send` endpoint not being able to create outlier events.
```
Comment (richvdh) [inline changelog.d/10938.bugfix:1]: we should say which version the bug was introduced in:

```suggestion
Fix bug introduced in Synapse 1.44 which caused the experimental [MSC2716](https://github.com/matrix-org/matrix-doc/pull/2716) `/batch_send` endpoint to return a 500 error.
```

(can we be more specific? "create outlier events" is too opaque for an admin - what causes it to generate outliers?)

**Review by richvdh — CHANGES_REQUESTED** [review]
Sorry for missing this one, @MadLittleMods. I didn't realise we had code in the MessageHandler which generated outliers :/

**Comment by MadLittleMods** [comment]
Thanks for the review pass @richvdh! It's hard to see when the MSC2716 tests don't run in CI yet 🙂

**Review by richvdh — APPROVED** [review]
thanks!

**Comment by reivilibre** [comment]
Merged #10938 into release-v1.44.

**Comment by MadLittleMods** [comment]
Thanks for the review @richvdh and for the merge @reivilibre 🦜

**Pull request closed without merging**